### DISCUSSION TITLE
v1.2.0b4

### DISCUSSION THREAD

**Discussion by JosephHewitt** [JosephHewitt/wardriver_rev3#115] [Announcements]
### Beta Release version 1.2.0b4

This version is a beta and is not yet considered to be stable.

### Changes since 1.2.0b3:

* Fix hash generation on B by @CoD-Segfault in https://github.com/JosephHewitt/wardriver_rev3/pull/98
* BW16 5GHz mod support for Rev3 (documentation still pending) in https://github.com/JosephHewitt/wardriver_rev3/pull/99
* Fix HDOP display on screen by @CoD-Segfault in https://github.com/JosephHewitt/wardriver_rev3/pull/100
* Improve documentation regarding version numbers in https://github.com/JosephHewitt/wardriver_rev3/pull/108
* Stop clearing MAC history on side B https://github.com/JosephHewitt/wardriver_rev3/pull/109
* Close files when done in https://github.com/JosephHewitt/wardriver_rev3/pull/112
* Update dependencies/board version in https://github.com/JosephHewitt/wardriver_rev3/pull/113
* Implement direct WiGLE uploading* in https://github.com/JosephHewitt/wardriver_rev3/pull/114

### Direct WiGLE upload feature*

You can now directly upload your captures to WiGLE and see basic stats about uploaded captures within the web interface of the Wardriver. This feature requires an API key from your WiGLE account which can be found at wigle.net/account and then by clicking "show token". Be sure to copy the one stating "Encoded for use" in full.

### Important note for users on previous betas:

Multiple issues have been identified in each version which can cause OTA update problems. You may see "Hash mismatch failed" when attempting to update the firmware of "B". This problem could potentially go away with future releases and may not affect everybody. You may install this update manually if this problem persists.

### Important note for users on v1.2.0b1:

Version v1.2.0b1 contains a bug which prevents OTA updates from working. To install this update, either perform it manually (using Arduino IDE or esptool), or follow the workaround documented in #86 

An additional bug in 1.2.0b1 could cause invalid checksums to be generated, this may mark updates as invalid even when they are not. Both of these issues are fixed in b2 and this release.

### Known issues

- You may not be able to update to a non-beta release over-the-air automatically.
- Release candidate versions may not be discovered correctly by the over-the-air functionality.
- Device boot may be slower if `cfg.txt` is on the SD card.
- Updates to B may indicate a newer version is available, even when there isn't.

<hr /><em>This discussion was created from the release <a href='https://github.com/JosephHewitt/wardriver_rev3/releases/tag/v1.2.0b4'>v1.2.0b4</a>.</em>

**Comment by pejacoby** [comment]
On my Rev3 WiSpy I had the B-side OTA update issue - hash failed message at the end of the installation, even though the various messages appeared to show the hashes fully matched. After manually flashing A and B with b4, I got the message that there was an update available until I removed the stray B.bin file from my SD card.

Otherwise running great so far, Wigle upload worked without a hitch!  Amazing update!
